src/Uno.UI/DataBinding/BindingPropertyHelper.CacheKeys.cs

```diff
@@ -0,0 +1,153 @@
+﻿#nullable enable
+
+#if !NETFX_CORE
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.UI.Xaml;
+
+namespace Uno.UI.DataBinding;
+
+internal static partial class BindingPropertyHelper
+{
+
+	private class GetValueGetterCacheKey(Type type, string name, DependencyPropertyValuePrecedences? precedence, bool allowPrivateMembers)
+	{
+		private readonly int _hashCode =
+				type.GetHashCode()
+				^ name.GetHashCode()
+				^ (precedence is not null ? (int)precedence : 0)
+				^ (allowPrivateMembers ? 1 : 0);
+
+		public Type Type { get; } = type;
+		public string Name { get; } = name;
+		public DependencyPropertyValuePrecedences? Precedence { get; } = precedence;
+		public bool PrivateMembers { get; } = allowPrivateMembers;
+
+		internal static IEqualityComparer<GetValueGetterCacheKey> Comparer { get; } = new EqualityComparer();
+
+		public override int GetHashCode() => _hashCode;
+
+		private class EqualityComparer : IEqualityComparer<GetValueGetterCacheKey>
+		{
+			public bool Equals(GetValueGetterCacheKey? x, GetValueGetterCacheKey? y)
+			{
+				if (x is not null && y is not null)
```

Comment:
Do we expect this code path to be hit with `x` and `y` being the same reference? In this case, doing an early `ReferenceEquals` check is likely to be faster.